### PR TITLE
fix: correct packaging issues

### LINKED ISSUE #55: wheels are not correctly built
When I then go to use mt2:
```
(coffea-dev) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % conda create -y --name test_env_5 numpy=1.23.5 python=3.9
(coffea-dev) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % conda activate test_env_5
(test_env_5) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % pip install mt2
Collecting mt2
  Downloading mt2-1.2.1-cp39-cp39-macosx_11_0_arm64.whl.metadata (7.6 kB)
Requirement already satisfied: numpy>=1.19.3 in /Users/lgray/miniforge3/envs/test_env_5/lib/python3.9/site-packages (from mt2) (1.23.5)
Downloading mt2-1.2.1-cp39-cp39-macosx_11_0_arm64.whl (22 kB)
Installing collected packages: mt2
Successfully installed mt2-1.2.1
(test_env_5) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % python
Python 3.9.19 | packaged by conda-forge | (main, Mar 20 2024, 12:55:20) 
[Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
im>>> import mt2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'mt2'
>>> 
(test_env_5) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % ls /Users/lgray/miniforge3/envs/test_env_5/lib/python3.9/site-packages
README.txt				distutils-precedence.pth		numpy-1.23.5.dist-info			pkg_resources				wheel
_distutils_hack				mt2-1.2.1.dist-info			pip					setuptools				wheel-0.44.0.dist-info
_mt2.cpython-39-darwin.so		numpy					pip-24.2.dist-info			setuptools-72.2.0-py3.12.egg-info
```

So something's not *quite* right in the way the wheels are being generated. The mt2 shared object library is being installed into the root of site-packages!

### PR DESCRIPTION
**Pull request by henryiii**
Should fix #55.

### CODE REVIEW

**Comment by tpgillam** [comment]
Thanks for looking at this!

I was initially surprised as I had hoped that the cibuildwheel option to test the wheels that we are using would have caught this, but it did not. We’ll need to figure out how to add a check that would have caught this.

**Comment by henryiii** [comment]
The problem is that you aren't using `src` structure. This is a good idea for all code, but an especially good idea for compiled code. See https://learn.scientific-python.org/development/guides/packaging-simple/#package-structure. So you are often picking up the local "mt2" directory and not noticing that it's not in the wheel.

File: setup.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 setup(
     ext_modules=[
         Extension(
-            "_mt2",
+            "mt2._mt2",
```
Comment (henryiii) [inline setup.py:10]:
```suggestion
            "_mt2",
```

The problem was that the `mt2` directory was missing, not that this was in the wrong place.

**Comment by tpgillam** [comment]
> The problem is that you aren't using `src` structure.

I can certainly believe that might be a fix.

But why must it be the _only_ fix? Another option would be for the tests to be run in an isolated environment in which we install the generated wheel explicitly. This seems less sensitive to future changes.

**Review by tpgillam — APPROVED** [review]

**Pull request merged**